### ISSUE TITLE
your documentation need to be updated to import and parse the correct flutter engine class

### ISSUE THREAD

**Issue by abdulmalekDery** [FirebaseExtended/flutterfire#3117]
this should be the correct syntax of the application class

package YOUR.PACKAGE.NAME;

import io.flutter.app.FlutterApplication;
import io.flutter.plugin.common.PluginRegistry;
import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback;
import io.flutter.plugins.GeneratedPluginRegistrant;
import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService;
import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin;
import io.flutter.embedding.engine.FlutterEngine;//<-- the correct import here

public class Application extends FlutterApplication implements PluginRegistrantCallback {
  @Override
  public void onCreate() {
    super.onCreate();
    FlutterFirebaseMessagingService.setPluginRegistrant(this);
  }

  @Override
  public void registerWith(PluginRegistry registry) {
    GeneratedPluginRegistrant.registerWith((FlutterEngine) registry);//<-- the correct parse here
  }
}

**Comment by darshankawar** [comment]
Currently in `readme.md`, it has:

```
@Override
  public void registerWith(PluginRegistry registry) {
    GeneratedPluginRegistrant.registerWith(registry);
  }
```

**Comment by abdulmalekDery** [comment]
@darshankawar 
it's not working
the project is not even running
I just found the best solution 
https://stackoverflow.com/a/60958130/6118808

**Comment by omishah** [comment]
Yes, you're absolutely correct. Today itself was following those old dead steps as available in the package page and ended up with errors. `readme.md` the file is so outdated and even `example` app is so dead. Kindly, update it. It's so confusing for us, at least for me.
